File: test/cpp/testpoint.cpp
#include <stdio.h>
#include <time.h>
#include <iostream>

using namespace std;

void test1(int32_t* p0,int len);
void test2(int32_t* p0,int len);

int main()
{
    printf("Hello world C++\r\n");

    int len = 1000000;
    int32_t* p0 = new int[len];

    clock_t start,end;


    start = clock();
    test1(p0,len);
    //需要测试运行时间的程序段
    end = clock();
    cout<<"test1运行时间(s): "<<(double)(end-start)/CLOCKS_PER_SEC<<endl;



    start = clock();
    test2(p0,len);
    //需要测试运行时间的程序段
    end = clock();
    cout<<"test2运行时间(s): "<<(double)(end-start)/CLOCKS_PER_SEC<<endl;


    printf("Bye world\r\n");
    return 0;
}

void test1(int32_t* p0,int len)
{
    for(int k=0;k<5000;k++){
        int32_t* p= p0 +1;
        int32_t* pEnd = p + len;
        while(p < pEnd){
            *p = k;
            p++;
        }
    }
}
void test2(int32_t* p0,int len)
{
    for(int k=0;k<5000;k++){
        int32_t* p= p0 +1;
        int32_t* pEnd = p + len;
        while(p < pEnd){
            *p = *(p-1);
            p++;
        }
    }
}

File: test/java/testpoint.java
public class testpoint{
    public static void main(String[] args){
        System.out.println("Hello World Java");
        int len = 1000000;
        int loops = 5000;
        int[] a = new int[len];
        long time1=System.currentTimeMillis();	
        testpoint1(a,loops);
        long time2=System.currentTimeMillis();	
        System.out.println("test1运行时间(ms): "+(double)((time2-time1)));
        time1=System.currentTimeMillis();	
        testpoint2(a,loops);
        time2=System.currentTimeMillis();	
        System.out.println("test2运行时间(ms): "+(time2-time1));
        time1=System.currentTimeMillis();	
        testpoint11(a,loops,1);
        time2=System.currentTimeMillis();	
        System.out.println("test11运行时间(ms): "+(time2-time1));
        time1=System.currentTimeMillis();	
        testpoint22(a,loops,1);
        time2=System.currentTimeMillis();	
        System.out.println("test22运行时间(ms): "+(time2-time1));
        System.out.println("Bye World Java");
    }

    public static void testpoint1(int[] array,int loops){ 
        for(int k =0;k< loops;k++)
        {
            for(int i=1;i< array.length;i++)
            {
                array[i] = k;
            }
        }

    }
    public static void testpoint2(int[] array,int loops){ 
        for(int k =0;k< loops;k++)
        {
            for(int i=1;i< array.length;i++)
            {
                array[i] = array[i-1];
            }
        }

    }
    public static void testpoint22(int[] array,int loops,int step){ 
        for(int k =0;k< loops;k++)
        {
            for(int i=1;i< array.length;i+=step)
            {
                array[i] = array[i-1];
            }
        }

    }
    public static void testpoint11(int[] array,int loops,int step){ 
        for(int k =0;k< loops;k++)
        {
            for(int i=1;i< array.length;i+=step)
            {
                array[i] = k;
            }
        }

    }
}